### PR TITLE
[MD] Address UX comments on index pattern stack

### PR DESCRIPTION
**Pull request by kristenTian**
- Update data source column header in index pattern table
- Update index pattern column name to Title
- Remove data source search field error check given it is not required

Signed-off-by: Kristen Tian <tyarong@amazon.com>

### Description
image

image

### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by kavilla** [comment]
Looks like a snapshot failure?

**Comment by KrooshalUX** [comment]
LGTM with one slight change -  the table column header should not appear sentence case and should read Data Source Connection

**Review by zhongnansu — APPROVED** [review]
lgtm

**Comment by zhongnansu** [comment]
@kristenTian can you change the title to something meaningful? So when we squash and merge. The commit message makes sense for user.

**Review by zhongnansu — APPROVED** [review]

File: src/plugins/index_pattern_management/public/components/index_pattern_table/index_pattern_table.tsx [hunk 1]
```diff
@@ -183,7 +183,7 @@ export const IndexPatternTable = ({ canSave, history }: Props) => {
   const columns = [
     {
       field: 'title',
-      name: 'Pattern',
+      name: 'Title',
```
Comment (kavilla) [inline src/plugins/index_pattern_management/public/components/index_pattern_table/index_pattern_table.tsx:186]: Do we have insight or if this favorable or not?

**Review by kavilla — CHANGES_REQUESTED** [review]
Just blocking for more insight on the `Pattern` to `Title` change

File: src/plugins/index_pattern_management/public/components/index_pattern_table/index_pattern_table.tsx (hunk 1 above)
Comment (kristenTian) [reply src/plugins/index_pattern_management/public/components/index_pattern_table/index_pattern_table.tsx:186]: @KrooshalUX requested this change. She mentioned to be consistent with save object page.

**Comment by kristenTian** [comment]
> @kristenTian can you change the title to something meaningful? So when we squash and merge. The commit message makes sense for user.

I have included all changes the commit message already. So even with squash, it will be visible to user

**Comment by KrooshalUX** [comment]
@kavilla As we were working through all of the interfaces, we determined that Index Pattern tables were inconsistent with the pattern across Dashboards, which is to use "Title", as it also reflects the object API.

**Comment by AMoo-Miki** [comment]
I would argue that this interface does not show a "title" or even a "pattern"; it is specifically an "index pattern". While creating an index pattern, you don't get to set a "name" or "title"; you only get to set an "index pattern".

PS, the index pattern creation page incorrectly calls it "Index pattern name" for a field that is really the "index pattern" and is not a name. We will have to fix that separately.

**Comment by kristenTian** [comment]
> I would argue that this interface does not show a "title" or even a "pattern"; it is specifically an "index pattern". While creating an index pattern, you don't get to set a "name" or "title"; you only get to set an "index pattern".
> 
> PS, the index pattern creation page incorrectly calls it "Index pattern name" for a field that is really the "index pattern" and is not a name. We will have to fix that separately.

This make sense to me, however still requires a further alignment on UX. Given the PR contains other small issues to address, I will revert the column title change for now and address it in a separate PR.

**Comment by KrooshalUX** [comment]
We can move forward without the change to Pattern / Title - given the discussion around naming and patterns that needs to happen at a larger scale.

File: src/plugins/index_pattern_management/public/components/index_pattern_table/index_pattern_table.tsx (hunk 1 above)
Comment (joshuarrrr) [reply src/plugins/index_pattern_management/public/components/index_pattern_table/index_pattern_table.tsx:186]: @kristenTian @KrooshalUX - We discussed in the brainstorming meeting today - the "title" reference in the saved object table is a bit misleading, and "pattern" or "index pattern" is the more accurate term. One potential solution in the future would be to disambiguate the index pattern saved object title from the pattern itself: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/2649

**Review by zhongnansu — APPROVED** [review]
lgtm

**Review by zengyan-amazon — APPROVED** [review]

**Pull request merged**